src/components/World/NodeSettings.tsx

```diff
@@ -966,16 +879,114 @@ class NodeSettings extends React.PureComponent<Props, State> {
     const robotOptions: ComboBox.Option[] = Dict.toList(robots)
       .map(([id, robot]) => ([id, Async.latestValue(robot)]))
       .filter(([, robot]: [string, Robot]) => robot !== undefined)
-      .map(([id, robot]: [string, Robot]) => ComboBox.option(LocalizedString.lookup(robot.name, LocalizedString.EN_US), id));
+      .map(([id, robot]: [string, Robot]) => ComboBox.option(LocalizedString.lookup(robot.name, locale), id));
+
+    const GEOMETRY_OPTIONS: ComboBox.Option[] = [
+      ComboBox.option(LocalizedString.lookup(tr('Box'), locale), 'box'),
+      ComboBox.option(LocalizedString.lookup(tr('Sphere'), locale), 'sphere'),
+      ComboBox.option(LocalizedString.lookup(tr('Cylinder'), locale), 'cylinder'),
+      ComboBox.option(LocalizedString.lookup(tr('Cone'), locale), 'cone'),
+      ComboBox.option(LocalizedString.lookup(tr('Plane'), locale), 'plane'),
+      ComboBox.option(LocalizedString.lookup(tr('File'), locale), 'file'),
+    ];
+    
+    const GEOMETRY_REVERSE_OPTIONS: Dict<number> = GEOMETRY_OPTIONS.reduce((dict, option, i) => {
+      dict[option.data as string] = i;
+      return dict;
+    }, {});
+    
+    
+    const TEMPLATE_OPTIONS: ComboBox.Option[] = [
+      ComboBox.option(LocalizedString.lookup(tr('Can'), locale), 'can'),
+      ComboBox.option(LocalizedString.lookup(tr('Paper Ream'), locale), 'ream'),
+    ];
+    
+    const TEMPLATE_REVERSE_OPTIONS: Dict<number> = TEMPLATE_OPTIONS.reduce((dict, option, i) => {
+      dict[option.data as string] = i;
+      return dict;
+    }, {});
+    
+    const ROTATION_TYPES: ComboBox.Option[] = [
+      ComboBox.option(LocalizedString.lookup(tr('Euler'), locale), 'euler'),
+      ComboBox.option(LocalizedString.lookup(tr('Axis Angle'), locale), 'angle-axis'),
+    ];
+    
+    const EULER_ORDER_OPTIONS: ComboBox.Option[] = [
+      ComboBox.option('XYZ', 'xyz'),
+      ComboBox.option('YZX', 'yzx'),
+      ComboBox.option('ZXY', 'zxy'),
+      ComboBox.option('XZY', 'xzy'),
+      ComboBox.option('YXZ', 'yxz'),
+      ComboBox.option('ZYX', 'zyx'),
```

Comment:
needs `tr()`... I think? I'm not sure how it'd get translated, but we have 'X' 'Y' and 'Z' translated elsewhere to might as well